### ISSUE TITLE
1.16.2 Crash cause of LazyOptional

### ISSUE THREAD

**Issue by BedrockLegends** [gigaherz/ToolBelt#78]
https://gist.github.com/BedrockLegends/3d6c7c25374703ea9c3fedb26d01edb3

**Comment by gigaherz** [comment]
Yep thanks for reminding me. I'll try to fix later.

**Comment by BedrockLegends** [comment]
Your welcome

**Comment by alorg** [comment]
I got the same issue, looking forward to the fixed version to try it on 1.16.2

**Comment by alorg** [comment]
[crash-2020-08-27_12.36.57-client.txt](https://github.com/gigaherz/ToolBelt/files/5136101/crash-2020-08-27_12.36.57-client.txt)
Here is my crash log if you need it.

**Comment by gigaherz** [comment]
Thanks. I didn't need it though. I was just too tired last night.
I took the chance to to a little cleanup of that code. It basically just needed to be recompiled, the same exact code I had would have worked even if it resulted in different binary due to the different object type, but the change I made to LazyOptional means I could avoid some ugly code, so yeah.
I just have to test a few things to make sure I didn't break the mod.
